pkg/node/manager/manager.go

```diff
@@ -734,6 +734,49 @@ func (m *manager) NodeUpdated(n nodeTypes.Node) {
 	m.ipsetMgr.AddToIPSet(ipset.CiliumNodeIPSetV4, ipset.INetFamily, v4Addrs...)
 	m.ipsetMgr.AddToIPSet(ipset.CiliumNodeIPSetV6, ipset.INet6Family, v6Addrs...)
 
+	if !n.IsLocal() {
+		for _, cidr := range n.GetIPv4AllocCIDRs() {
+			if cidr == nil {
+				continue
+			}
+
+			prefix, ok := netipx.FromStdIPNet(cidr.IPNet)
+			if !ok {
+				continue
+			}
+
+			metadata := []ipcache.IPMetadata{
+				ipcacheTypes.RequestedIdentity(identity.ReservedIdentityWorldIPv4),
```

Comment:
I believe this needs to use `GetWorldIdentityFromIP` for non-dual-stack setups

Reply:
Well, as it turns out (see below), this is actually completely ignored, which is probably why CI was passing